From: Author <author@example.com>
Date: Thu, 1 Jan 1970 00:00:00 +0000
Subject: [PATCH] add build file

---
 BUILD.bazel | 63 +++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 63 insertions(+)
 create mode 100644 BUILD.bazel

diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..f8f2d8e
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,63 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+load("@rules_license//rules:license.bzl", "license")
+load("@rules_license//rules:package_info.bzl", "package_info")
+
+package(
+    default_applicable_licenses = [
+        ":license",
+        ":package_info",
+    ],
+    default_visibility = ["//visibility:public"],
+)
+
+license(
+    name = "license",
+    license_kinds = [
+        "@rules_license//licenses/spdx:LGPL-3.0-only",
+    ],
+    license_text = "COPYING",
+)
+
+package_info(
+    name = "package_info",
+    package_name = "libheif",
+    cpe = "cpe:2.3:a:struktur:libheif",
+    package_version = "1.18.2",
+)
+cmake(
+    name = "libheif",
+    cache_entries = CACHE_ENTRIES,
+    copts = [
+        "-x none",
+    ],
+    lib_source = ":srcs",
+    out_shared_libs = select({
+        "@platforms//os:macos": ["libheif.dylib"],
+        "//conditions:default": ["libheif.so.1"],
+    }),
+    deps = [
+        "@libaom",
+        "@libde265",
+    ],
+)
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+)
-- 
2.49.0